pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/pubsub/MercuryOrderStatusListener.java

```diff
@@ -0,0 +1,64 @@
+package org.broadinstitute.dsm.pubsub;
+
+import java.sql.Connection;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import com.google.api.gax.core.ExecutorProvider;
+import com.google.api.gax.core.InstantiatingExecutorProvider;
+import com.google.cloud.pubsub.v1.AckReplyConsumer;
+import com.google.cloud.pubsub.v1.MessageReceiver;
+import com.google.cloud.pubsub.v1.Subscriber;
+import com.google.gson.Gson;
+import com.google.pubsub.v1.ProjectSubscriptionName;
+import com.google.pubsub.v1.PubsubMessage;
+import org.broadinstitute.ddp.db.TransactionWrapper;
+import org.broadinstitute.dsm.db.dao.mercury.MercuryOrderDao;
+import org.broadinstitute.dsm.model.mercury.BaseMercuryStatusMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MercuryOrderStatusListener {
+    private static final Logger logger = LoggerFactory.getLogger(DSMtasksSubscription.class);
+
+    public static void subscribeToOrderStatus(String projectId, String subscriptionId) {
+        // Instantiate an asynchronous message receiver.
+        MessageReceiver receiver = (PubsubMessage message, AckReplyConsumer consumer) -> {
+            // Handle incoming message, then ack the received message.
+            logger.info("Got STATUS message with Id: " + message.getMessageId());
+
+            try {
+                TransactionWrapper.inTransaction(conn -> {
```

Comment:
Instead of having transaction here, why not to use dao class directly and you won't have to pass the connection as well